docs/guide/dev/features/autocomplete/README.md

```diff
@@ -14,4 +14,140 @@ For licensing, see LICENSE.md.
+
+## Usage
+
+The autocomplete plugin shows dropdown with available options every time when user types matching text.
+
+{@img assets/img/autocomplete_01.png Using autocomplete to get ticket hints.}
+
+When you press <kbd>enter</kbd>, <kbd>tab</kbd> or any other customized [`commitKeystroke`](https://docs.ckeditor.com/ckeditor4/docs/#!/api/CKEDITOR.config#cfg-autocomplete_commitKeystrokes) suggested value will be inserted into editor. 
+
+## Configuration
+
+Autocomplete plugin utilizes two important callbacks which allows you to customize matching function and data source:
+
+* A text test callback.
+* A data callback.
+
+They are required to setup autocomplete instance which will be immediately attached into editor after its creation. Autocomplete can be configured by [`configDefinition`](https://docs.ckeditor.com/ckeditor4/docs/#!/api/CKEDITOR_plugins_autocomplete.configDefition) object passed into autocomplete constructor.
+
+```javascript
+// We will update this object during this guide.
+var config = {};
+```
+
+## Text test callback
+
+A function which should return a fragment of text (typed in the editor) that should be autocompleted. This function works best with [`textmatch`](https://docs.ckeditor.com/ckeditor4/docs/#!/api/CKEDITOR_plugins_textwatcher) feature which was published alongside with `autocomplete` plugin. Lets say you would like to create autocompletion feature for GitHub tickets. Depending on you use case, you could create autocomplete instance using this code:
+
+```javascript
+// Called when the user types in the editor or moves the caret.
+// The range represents the caret position.
+function textTestCallback( range ) {
+	// We don't want to autocomplete a non-empty selection.
+	if ( !range.collapsed ) {
+		return null;
+	}
+
+	// Use the textmatch plugin which does the tricky job of doing
+	// a text search in the DOM. The matchCallback function should return
+	// a matching fragment of the text.
+	return CKEDITOR.plugins.textMatch.match( range, matchCallback );
+}
+
+// Returns a position of the matching text.
+// It matches with text starting from the '#' character
+// followed by spaces, up to the caret position.
+function matchCallback( text, offset ) {
+	// Get the text before the caret.
+	var left = text.slice( 0, offset ),
+		// Will look for an '#' character followed by word characters.
+		match = left.match( /#\w$/ );
+
+	if ( !match ) {
+		return null;
+	}
+	return { start: match.index, end: offset };
+}
+
+config.textTestCallback = textTestCallback;
+```
+
+## Data callback
+
+A function which should return (through its callback) a suggestion data for the current query string. This function will be only called if the previous `textTestCallback` returned matching text.
+
+```javascript
+// The itemsArray variable is our example "database".
+var itemsArray = [
+	{ id: 1, name: 'Introduce autocomplete feature', ticket: '134' },
```

Comment:
Technically `id` field is redundant in this particular sample, as `ticket` field should also be unique…

So my suggestion is to move `ticket` value to id (as a number). Then in the line where you do filtering:

```js
var suggestions = itemsArray.filter( function( item ) {
        return item.ticket.indexOf( query ) === 0;
    } );
```

You could filter based on id, by casting it to `String`, like so:

```js
return String( item.id ).indexOf( query ) === 0;
```

This will also simplify the sample 👍

Reply:
Make sure to update other parts of guide accordingly after doing the above. ☝️